README.md
# BrawlQuest
## THIS IS OLD, PRE-UNIVERSITY EDUCATION CODE. IT'S GARBAGE. IT'S TERRIBLE. PLEASE DON'T USE IT OR READ IT.
### But it's public now so do what you want


changelog.txt
TODO: login password section displays player name as entered password, logs in with it

Alpha v1.3
##Content
* Raid opened: Crypt of the Sorcerer
	* Two phase finale fight
	* Rewards 700 Quia, 50% chance "Dragonling", 10% chance "Eldertouched Plate"
* New armour piece: Eldertouched Plate
	* Red aura
	* 50 DEF
* NPCs will no longer attack invincible targets

##Spells
* Recovery now heals 100% of your health and has had its cooldown reduced to 30 seconds (down from 120 seconds)
* Slam now deals twice as much damage (ATK*8) and has a 20 second cooldown (down from 30 seconds)

##UI
* Added new character select screen
	* To login as a character that already exists, press "Add New" and enter your username and password
* Added chat
	* Press enter and start typing, when you press enter again the message will be sent
	* Currently all chat is global
* Added floating combat text
* FreshPlay logo is now displayed on the splash screen
* ESC now switches between fullscreen / windowed (rather than F)
* NPC dialogue is now centred regardless of zoom level
* Fixed bones - enjoy the chaos!!
* "Earned Loot" window no longer displays for loot losses (such as the gold lost when purchasing an item)

##Misc
* Player buddies are now only displayed if the player is online
* Buddies are now drawn underneath their owners
* Website character profiles now display attack bonuses from Orbs of Power
* Added old theme tune back into rotation
* Zooming is now at set levels
* The server now tracks last login and displays this on HTML character profiles
* New voice lines

Alpha v1.2
##Graphics & UI
* Changed the sprite for Basic Cloth
* Changed the sprite for Dog
* Nameplates are now darker
* The Blacksmith now has "<NPC> Shop" above his name
* The "Pray" window no longer shows up on your current spawn point (this also acts as temporary visual feedback that the spawn point is set)
* Pressing F now switches between fullscreen and windowed mode
* Several changes and fixes to "New Loot"

##Misc
* Reduced number of world updates to increase FPS
* Allowed the server to send data on maximum HP allowing HP changes without requiring client updates
* Increased accuracy of playtime reporting
* Server now tracks kills, deaths and distance travelled
* You can no longer see offline players in the overworld

##Content
* New zone: Isle of Giants
	* Features 3 miniboss fights that each reward 500 Adver
	* Features 1 boss fight that combines all 3 other fights that rewards 200 Quia
* Added 4 secret pets
* You can now exchange 5 Adver for 1 Quia at the shop
* The shop outside the dungeon "Prison of the Sorcerer" is now open for business!
* Increased Skeleton King's HP from 6000 to 12000
	* Increased Skeleton King's venom spawn time from 5s to 20s
* Increased Hydra's HP from 4000 to 6000
* Made Mortus invincible
* Reduced Guard's HP from 60 to 30


mobs.lua
mb = {} --identified by name
mb.hp = {}
mb.spd = {}
mb.atk = {}
mb.sp1 = {} --spell 1
mb.sp1t = {}
mb.sp2 = {} --spell 2
mb.sp2t = {}
mb.img = {} --width in this case
mb.rng = {}
mb.sfx = {}
mb.friend = {}

local tm = "Player" --for bones
mb.img[tm] = love.graphics.newImage("img/dragon/Red.png")

mb.img["Mortus"] = love.graphics.newImage("img/human/Mortus.png")

local tm = "Adventurer"
mb.hp[tm] = 1
mb.spd[tm] = 50
mb.atk[tm] = 1
mb.sp1[tm] = "None"
mb.sp1t[tm] = 0
mb.sp2[tm] = "None"
mb.sp2t[tm] = 0
mb.img[tm] = love.graphics.newImage("img/human/Adventurer.png")
mb.rng[tm] = 32
mb.friend[tm] = true

local tm = "Guard"
mb.hp[tm] = 5
mb.spd[tm] = 64
mb.atk[tm] = 1
mb.sp1[tm] = "suicide"
mb.sp1t[tm] = 5
mb.sp2[tm] = "None"
mb.sp2t[tm] = 0
mb.img[tm] = love.graphics.newImage("img/human/Guard.png")
mb.rng[tm] = 64
mb.friend[tm] = true

local tm = "Mortus"
mb.hp[tm] = 100
mb.spd[tm] = 128
mb.atk[tm] = 4
mb.sp1[tm] = "None"
mb.sp1t[tm] = 0
mb.sp2[tm] = "None"
mb.sp2t[tm] = 0
mb.img[tm] = love.graphics.newImage("img/human/Mortus.png")
mb.rng[tm] = 60
mb.friend[tm] = true

local tm = "Angry Chicken"
mb.hp[tm] = 1
mb.spd[tm] = 128
mb.atk[tm] = 2
mb.sp1[tm] = "Suicide"
mb.sp1t[tm] = 5
mb.sp2[tm] = "None"
mb.sp2t[tm] = 0
mb.img[tm] = love.graphics.newImage("img/mob/Chicken.png")
mb.rng[tm] = 16
mb.friend[tm] = true

local tm = "Boar"
mb.hp[tm] = 1
mb.spd[tm] = 64
mb.atk[tm] = 2
mb.sp1[tm] = "None"
mb.sp1t[tm] = 0
mb.sp2[tm] = "None"
mb.sp2t[tm] = 0
mb.img[tm] = love.graphics.newImage("img/mob/Boar.png")
mb.rng[tm] = 16
mb.sfx[tm] = {}
  mb.sfx[tm][1] = love.audio.newSource("sound/sfx/mob/Boar/1.ogg")
  mb.sfx[tm][2] = love.audio.newSource("sound/sfx/mob/Boar/2.ogg")
  mb.sfx[tm][3] = love.audio.newSource("sound/sfx/mob/Boar/3.ogg")
  mb.sfx[tm][4] = love.audio.newSource("sound/sfx/mob/Boar/4.ogg")

  local tm = "Big Boar"
  mb.hp[tm] = 5
  mb.spd[tm] = 35
  mb.atk[tm] = 4
  mb.sp1[tm] = "None"
  mb.sp1t[tm] = 0
  mb.sp2[tm] = "None"
  mb.sp2t[tm] = 0
mb.img[tm] = love.graphics.newImage("img/mob/Big Boar.png")
mb.rng[tm] = 16
mb.sfx[tm] = {}
  mb.sfx[tm][1] = love.audio.newSource("sound/sfx/mob/Boar/1.ogg")
  mb.sfx[tm][2] = love.audio.newSource("sound/sfx/mob/Boar/2.ogg")
  mb.sfx[tm][3] = love.audio.newSource("sound/sfx/mob/Boar/3.ogg")
  mb.sfx[tm][4] = love.audio.newSource("sound/sfx/mob/Boar/4.ogg")

  local tm = "Biggest Boar"
    mb.hp[tm] = 20
    mb.spd[tm] = 16
    mb.atk[tm] = 50
    mb.sp1[tm] = "None"
    mb.sp1t[tm] = 0
    mb.sp2[tm] = "None"
    mb.sp2t[tm] = 0
  mb.img[tm] = love.graphics.newImage("img/mob/Biggest Boar.png")
  mb.rng[tm] = 16
  mb.sfx[tm] = {}
    mb.sfx[tm][1] = love.audio.newSource("sound/sfx/mob/Boar/1.ogg")
    mb.sfx[tm][2] = love.audio.newSource("sound/sfx/mob/Boar/2.ogg")
    mb.sfx[tm][3] = love.audio.newSource("sound/sfx/mob/Boar/3.ogg")
    mb.sfx[tm][4] = love.audio.newSource("sound/sfx/mob/Boar/4.ogg")

  local tm = "Ghost"
    mb.hp[tm] = 100
    mb.spd[tm] = 200
    mb.atk[tm] = 100
    mb.sp1[tm] = "None"
    mb.sp1t[tm] = 0
    mb.sp2[tm] = "None"
    mb.sp2t[tm] = 0
    mb.img[tm] = love.graphics.newImage("img/mob/Ghost.png")
    mb.rng[tm] = 16

    local tm = "Savage Wolf"
        mb.hp[tm] = 1
        mb.spd[tm] = 100
        mb.atk[tm] = 1
        mb.sp1[tm] = "None"
        mb.sp1t[tm] = 0
        mb.sp2[tm] = "None"
        mb.sp2t[tm] = 0
        mb.img[tm] = love.graphics.newImage("img/mob/Savage Wolf.png")
        mb.rng[tm] = 16

    local tm = "Wolf"
        mb.hp[tm] = 5
        mb.spd[tm] = 80
        mb.atk[tm] = 25
        mb.sp1[tm] = "None"
        mb.sp1t[tm] = 0
        mb.sp2[tm] = "None"
        mb.sp2t[tm] = 0
        mb.img[tm] = love.graphics.newImage("img/mob/Wolf.png")
        mb.rng[tm] = 16

        local tm = "Alpha Wolf"
            mb.hp[tm] = 20
            mb.spd[tm] = 65
            mb.atk[tm] = 30
            mb.sp1[tm] = "None"
            mb.sp1t[tm] = 0
            mb.sp2[tm] = "None"
            mb.sp2t[tm] = 0
            mb.img[tm] = love.graphics.newImage("img/mob/Alpha Wolf.png")
            mb.rng[tm] = 16

        local tm = "Mother Wolf"
          mb.hp[tm] = 20
            mb.spd[tm] = 40
              mb.atk[tm] = 30
                mb.sp1[tm] = "spawn:Wolf"
                mb.sp1t[tm] = 20
                mb.sp2[tm] = "None"
                mb.sp2t[tm] = 0
                mb.img[tm] = love.graphics.newImage("img/mob/Mother Wolf.png")
                mb.rng[tm] = 16

                local tm = "Cub"
                  mb.hp[tm] = 1
                  mb.spd[tm] = 120
                  mb.atk[tm] = 1
                  mb.sp1[tm] = "None"
                  mb.sp1t[tm] = 0
                  mb.sp2[tm] = "None"
                  mb.sp2t[tm] = 0
                  mb.img[tm] = love.graphics.newImage("img/mob/Cub.png")
                  mb.rng[tm] = 16

local tm = "Sorcerer"
mb.hp[tm] = 1
mb.spd[tm] = 64
mb.atk[tm] = 2
mb.sp1[tm] = "None"
mb.sp1t[tm] = 0
mb.sp2[tm] = "None"
mb.sp2t[tm] = 0
mb.img[tm] = love.graphics.newImage("img/mob/Sorcerer.png")
mb.rng[tm] = 16

local tm = "Cursed Human"
mb.hp[tm] = 1
mb.spd[tm] = 96
mb.atk[tm] = 2
mb.sp1[tm] = "None"
mb.sp1t[tm] = 0
mb.sp2[tm] = "None"
mb.sp2t[tm] = 0
mb.img[tm] = love.graphics.newImage("img/mob/"..tm..".png")
mb.rng[tm] =  32

local tm = "Cursed Guard"
mb.hp[tm] = 5
mb.spd[tm] = 70
mb.atk[tm] = 4
mb.sp1[tm] = "None"
mb.sp1t[tm] = 0
mb.sp2[tm] = "None"
mb.sp2t[tm] = 0
mb.img[tm] = love.graphics.newImage("img/mob/"..tm..".png")
mb.rng[tm] =  32

local tm = "Cursed King"
mb.hp[tm] = 25
mb.spd[tm] = 60
mb.atk[tm] = 8
mb.sp1[tm] = "spawn:Cursed Human"
mb.sp1t[tm] = 10
mb.sp2[tm] = "spawn:Cursed Guard"
mb.sp2t[tm] = 30
mb.img[tm] = love.graphics.newImage("img/mob/"..tm..".png")
mb.rng[tm] =  32

local tm = "Cursed Miner"
mb.hp[tm] = 15
mb.spd[tm] = 30
mb.atk[tm] = 10
mb.sp1[tm] = "None"
mb.sp1t[tm] = 0
mb.sp2[tm] = "None"
mb.sp2t[tm] = 0
mb.img[tm] = love.graphics.newImage("img/mob/"..tm..".png")
mb.rng[tm] =  32

local tm = "Savage Bat"
mb.hp[tm] = 3
mb.spd[tm] = 120
mb.atk[tm] = 2
mb.sp1[tm] = "None"
mb.sp1t[tm] = 0
mb.sp2[tm] = "None"
mb.sp2t[tm] = 0
mb.img[tm] = love.graphics.newImage("img/mob/"..tm..".png")
mb.rng[tm] = 16

local tm = "Savage Bear"
mb.hp[tm] = 30
mb.spd[tm] = 70
mb.atk[tm] = 5
mb.sp1[tm] = "None"
mb.sp1t[tm] = 0
mb.sp2[tm] = "None"
mb.sp2t[tm] = 0
mb.img[tm] = love.graphics.newImage("img/mob/"..tm..".png")
mb.rng[tm] = 16

local tm = "Savage Hydra"
mb.hp[tm] = 50
mb.spd[tm] = 60
mb.atk[tm] = 10
mb.sp1[tm] = "None"
mb.sp1t[tm] = 0
mb.sp2[tm] = "None"
mb.sp2t[tm] = 0
mb.img[tm] = love.graphics.newImage("img/mob/"..tm..".png")
mb.rng[tm] = 32

local tm = "Savage Golem"
mb.hp[tm] = 100
mb.spd[tm] = 40
mb.atk[tm] = 20
mb.sp1[tm] = "None"
mb.sp1t[tm] = 0
mb.sp2[tm] = "None"
mb.sp2t[tm] = 0
mb.img[tm] = love.graphics.newImage("img/mob/"..tm..".png")
mb.rng[tm] = 64

local tm = "Quake"
mb.hp[tm] = 100000
mb.spd[tm] = 0
mb.atk[tm] = 30
mb.sp1[tm] = "None"
mb.sp1t[tm] = 0
mb.sp2[tm] = "None"
mb.sp2t[tm] = 0
mb.img[tm] = love.graphics.newImage("img/mob/Quake.png")
mb.rng[tm] = 8
mb.friend[tm] = false

local tm = "Venom"
mb.hp[tm] = 100
mb.spd[tm] = 0
mb.atk[tm] = 0
mb.sp1[tm] = "dmg,20"
mb.sp1t[tm] = 5
mb.sp2[tm] = "None"
mb.sp2t[tm] = 0
mb.img[tm] = love.graphics.newImage("img/mob/Venom.png")
mb.rng[tm] = 0
mb.friend[tm] = false

local tm = "Savage Giant"
mb.hp[tm] = 4000
mb.spd[tm] = 18
mb.atk[tm] = 10
mb.sp1[tm] = "spawnFeet,Quake"
mb.sp1t[tm] = 5
mb.sp2[tm] = "None"
mb.sp2t[tm] = 0
mb.img[tm] = love.graphics.newImage("img/mob/Savage Giant.png")
mb.rng[tm] = 80
mb.friend[tm] = false

local tm = "Hydra"
mb.hp[tm] = 4000
mb.spd[tm] = 32
mb.atk[tm] = 20
mb.sp1[tm] = "spawn,Venom"
mb.sp1t[tm] = 5
mb.sp2[tm] = "spawn,Savage Hydra"
mb.sp2t[tm] = 15
mb.img[tm] = love.graphics.newImage("img/mob/Hydra.png")
mb.rng[tm] = 32
mb.friend[tm] = false

local tm = "Skeleton King"
mb.hp[tm] = 6000
mb.spd[tm] = 20
mb.atk[tm] = 15
mb.sp1[tm] = "spawnFeet,Venom"
mb.sp1t[tm] = 30
mb.sp2[tm] = "spawnFeet,Quake"
mb.sp2t[tm] = 20
mb.img[tm] = love.graphics.newImage("img/mob/Skeleton King.png")
mb.rng[tm] = 32
mb.friend[tm] = false

local tm = "The Skeleton King"
mb.img[tm] = love.graphics.newImage("img/mob/Mini Skeleton King.png")

local tm = "Giant"
mb.hp[tm] = 6000
mb.spd[tm] = 20
mb.atk[tm] = 15
mb.sp1[tm] = "spawnFeet,Quake"
mb.sp1t[tm] = 3
mb.sp2[tm] = "spawn,Golem"
mb.sp2t[tm] = 10
mb.img[tm] = love.graphics.newImage("img/mob/Giant.png")
mb.rng[tm] = 50
mb.friend[tm] = false

local tm = "Ice Giant"
mb.hp[tm] = 4000
mb.spd[tm] = 10
mb.atk[tm] = 5
mb.sp1[tm] = "spawnFeet,Ice"
mb.sp1t[tm] = 3
mb.sp2[tm] = "None"
mb.sp2t[tm] = 0
mb.img[tm] = love.graphics.newImage("img/mob/Ice Giant.png")
mb.rng[tm] = 50
mb.friend[tm] = false

local tm = "Ice"
mb.hp[tm] = 60000
mb.spd[tm] = 5
mb.atk[tm] = 30
mb.sp1[tm] = "None"
mb.sp1t[tm] = 30
mb.sp2[tm] = "None"
mb.sp2t[tm] = 20
mb.img[tm] = love.graphics.newImage("img/mob/Ice.png")
mb.rng[tm] = 32
mb.friend[tm] = false

local tm = "Fire Giant"
mb.hp[tm] = 2000
mb.spd[tm] = 30
mb.atk[tm] = 10
mb.sp1[tm] = "spawn,Fire"
mb.sp1t[tm] = 2
mb.sp2[tm] = "spawn,Fire"
mb.sp2t[tm] = 5
mb.img[tm] = love.graphics.newImage("img/mob/Fire Giant.png")
mb.rng[tm] = 50
mb.friend[tm] = false

local tm = "Fire"
mb.hp[tm] = 2000000
mb.spd[tm] = 120
mb.atk[tm] = 40
mb.sp1[tm] = "suicide"
mb.sp1t[tm] = 3
mb.sp2[tm] = "None"
mb.sp2t[tm] = 0
mb.img[tm] = love.graphics.newImage("img/mob/Fire.png")
mb.rng[tm] = 32
mb.friend[tm] = false
